src/ray/core_worker/reference_count.cc

```diff
@@ -1012,8 +1017,11 @@ bool ReferenceCounter::RemoveObjectLocation(const ObjectID &object_id,
                   << " that doesn't exist in the reference table";
     return false;
   }
-  it->second.locations.erase(node_id);
-  PushToLocationSubscribers(it);
+  it->second.locations[node_id]--;
```

Comment:
What if it goes negative due to out of order?

Reply:
It's definitely okay for this to be negative. The assumption is that eventually you will receive the corresponding Add request and then this will go back to 0.